crates/biome_js_analyze/tests/specs/style/noNegationElse/invalid.js.snap

```diff
@@ -0,0 +1,220 @@
+---
+source: crates/biome_js_analyze/tests/spec_tests.rs
+expression: invalid.js
+---
+# Input
+```js
+// Comment
+if (!true) { // a
+	consequent;
+} else { // b
+	alternate;
+}
+
+!condition ? consequent : alternate;
+
+let a = !test ? c : d;
+
+if (!true) {
+	consequent;
+} else something();
+
+if (!true) something(); else {
+	alternate
+}
+
+!-a ? b : c
+
+/*1*/!/*2*/cond/*3*/ ? /*a*/ b /*c*/ : /*d*/ e /*f*/
+
+```
+
+# Diagnostics
+```
+invalid.js:2:1 lint/style/noNegationElse  FIXABLE  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
+
+  ! Invert blocks when performing a negation test.
+  
+    1 │ // Comment
+  > 2 │ if (!true) { // a
+      │ ^^^^^^^^^^^^^^^^^
```

Comment:
I believe this range is incorrect, it's highlighting the comments too.

Reply:
In fact the entire `if` statement is covered by the range, including the inner trivia. Should I reduce to the condition?

Reply:
This is the trimmed range, `AstNode::range` uses `.syntax().trimmed_range()` under the hood. This comment is highlighted because it is part of the rimmed range that covers the entire statement. The only way to avoid it is to use a trimmed range of another node such as the test.

Reply:
Thank you for clarifying